huaweicloud/services/secmaster/resource_huaweicloud_secmaster_incident.go

```diff
@@ -0,0 +1,589 @@
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	id, err := jmespath.Search("data.id", createIncidentRespBody)
+	if err != nil {
+		return diag.Errorf("error creating Incident: ID is not found in API response")
+	}
+	d.SetId(id.(string))
+
+	return resourceIncidentRead(ctx, d, meta)
+}
+
+func buildIncidentBodyParams(d *schema.ResourceData, cfg *config.Config) (map[string]interface{}, error) {
+	dataObject := map[string]interface{}{
+		"title":              utils.ValueIngoreEmpty(d.Get("name")),
+		"description":        utils.ValueIngoreEmpty(d.Get("description")),
+		"incident_type":      buildIncidentRequestBodyType(d.Get("type")),
+		"severity":           utils.ValueIngoreEmpty(d.Get("level")),
+		"handle_status":      utils.ValueIngoreEmpty(d.Get("status")),
+		"owner":              utils.ValueIngoreEmpty(d.Get("owner")),
+		"data_source":        buildIncidentRequestBodyDataSource(d.Get("data_source")),
+		"verification_state": utils.ValueIngoreEmpty(d.Get("verification_status")),
+		"ipdrr_phase":        utils.ValueIngoreEmpty(d.Get("stage")),
+		"simulation":         utils.ValueIngoreEmpty(d.Get("debugging_data")),
+		"labels":             utils.ValueIngoreEmpty(d.Get("labels")),
+		"close_reason":       utils.ValueIngoreEmpty(d.Get("close_reason")),
+		"close_comment":      utils.ValueIngoreEmpty(d.Get("close_comment")),
+		"environment":        buildEnvironmentOpts(d, cfg),
+		"domain_id":          cfg.DomainID,
+		"region_id":          cfg.GetRegion(d),
+	}
+
+	inputTimeFormat := "2006-01-02T15:04:05.000-0700"
+
+	if v, ok := d.GetOk("first_occurrence_time"); ok {
+		firstOccurrenceTime, err := time.Parse(standardTimeFormat, v.(string))
+		if err != nil {
+			return nil, fmt.Errorf("error parsing first_occurrence_time: %s", err)
+		}
+
+		firstOccurrenceTimeWithoutZ := firstOccurrenceTime.Format(inputTimeFormat)
+		firstOccurrenceTimeWithZ := fmt.Sprintf(firstOccurrenceTimeWithoutZ[:23] + "Z" + firstOccurrenceTimeWithoutZ[23:])
+
+		dataObject["first_observed_time"] = firstOccurrenceTimeWithZ
+	}
+
+	if v, ok := d.GetOk("last_occurrence_time"); ok {
+		lastOccurrenceTime, err := time.Parse(standardTimeFormat, v.(string))
+		if err != nil {
+			return nil, fmt.Errorf("error parsing last_occurrence_time: %s", err)
+		}
+
+		lastOccurrenceTimeWithoutZ := lastOccurrenceTime.Format(inputTimeFormat)
+		lastOccurrenceTimeWithZ := fmt.Sprintf(lastOccurrenceTimeWithoutZ[:23] + "Z" + lastOccurrenceTimeWithoutZ[23:])
+
+		dataObject["last_observed_time"] = lastOccurrenceTimeWithZ
+	}
+
+	if v, ok := d.GetOk("planned_closure_time"); ok {
+		plannedClosureTime, err := time.Parse(standardTimeFormat, v.(string))
+		if err != nil {
+			return nil, fmt.Errorf("error parsing planned_closure_time: %s", err)
+		}
+
+		plannedClosureTimeWithoutZ := plannedClosureTime.Format(inputTimeFormat)
+		plannedClosureTimeWithZ := fmt.Sprintf(plannedClosureTimeWithoutZ[:23] + "Z" + plannedClosureTimeWithoutZ[23:])
```

Comment:
how about add a new function to convert format time.

Reply:
sure